### PR TITLE
11 get reviews queries

### PR DESCRIPTION
**Pull request by GWorsnop**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -233,6 +233,109 @@ describe("GET: /api/reviews", () => {
...
+          review_img_url:
+            "https://www.golenbock.com/wp-content/uploads/2015/01/placeholder-user.png",
+          title: "Agricola",
+          votes: 1,
+        });
+      });
+  });
+  test("400: bad request if sort_by category does not exist", () => {
+    return request(app)
+      .get("/api/reviews/?sort_by=George")
```
Comment (douglashellowell) [inline __tests__/app.test.js:325]: no need for the trailing slash on the endpoint here

```diff
- /api/reviews/?sort_by=George
+ /api/reviews?sort_by=George
```

File: app/models/reviews-model.js [hunk 2]
```diff
@@ -42,15 +42,85 @@ exports.updateReviewVotes = (inc_votes, review_id) => {
...
-  return connection
-    .query(
-      `
+exports.selectReviews = (query = { sort_by: "created_at" }) => {
+  let methods = ["sort_by", "order", "category"];
+  let searchSort_by = "created_at";
+  let searchOrder = "DESC";
+  let whereStr = "";
+  const request = Object.keys(query);
+  if (!methods.includes(request[0])) {
```
Comment (douglashellowell) [inline app/models/reviews-model.js:51]: why are you only checking the first query here?

what if it was:
```
GET /api/reviews?sort_by=votes&danger=zone
```
?

File: __tests__/app.test.js (hunk 1 above)
Comment (GWorsnop) [reply __tests__/app.test.js:325]: Edit to remove /

File: app/models/reviews-model.js (hunk 2 above)
Comment (GWorsnop) [reply app/models/reviews-model.js]: Edit to change to a for loop.  Had some issues with my promise.reject not preventing the rest of the code from working, so wrapped it all in a big if else.

**Review by douglashellowell — APPROVED** [review]
Looks great!

ready to merge

**Pull request merged**